### PR TITLE
Allow constructors with more than 1 parameter to be mocked

### PR DESCRIPTION
**Pull request by psaxton**
This is a fix for Issue #169.

### CODE REVIEW

File: test/index.test.js [hunk 1]
```diff
@@ -491,6 +491,14 @@ test('AWS.setSDK function should mock a specific AWS module', function(t) {
     });
   });
 
+  t.test('Modules with multi-parameter constructors can be set for mocking', function(st) {
+	awsMock.setSDK('aws-sdk');
+	awsMock.mock('CloudFront.Signer', 'getSignedUrl');
+	var signer = new AWS.CloudFront.Signer('key-pair-id', 'private-key');
+	st.ok(signer);
+	st.end();
```
Comment (abetomo) [inline test/index.test.js]: The width of the indent seems to be unified with 2.

Comment (psaxton) [reply test/index.test.js]: Updated for whitespacing.

**Comment by acchaulk** [comment]
is this ready to be merged?

**Comment by psaxton** [comment]
@acchaulk If you're asking me, then yes.

**Comment by psaxton** [comment]
@abetomo, @acchaulk can this be merged, please?

**Comment by abetomo** [comment]
@nelsonic Could you review it?

File: test/index.test.js [hunk 2]
```diff
@@ -491,6 +491,14 @@ test('AWS.setSDK function should mock a specific AWS module', function(t) {
     });
   });
 
+  t.test('Modules with multi-parameter constructors can be set for mocking', function(st) {
+	awsMock.setSDK('aws-sdk');
+	awsMock.mock('CloudFront.Signer', 'getSignedUrl');
+	var signer = new AWS.CloudFront.Signer('key-pair-id', 'private-key');
+	st.ok(signer);
```
Comment (nelsonic) [inline test/index.test.js]: `st.ok` checks for "_truthiness_" (_i.e. the existence of the variable_) not the actual value.
It's an "OK" assertion, but it's not a _precise_ one.

**Comment by nelsonic** [comment]
@psaxton `aws-sdk-mock@4.4.0` on NPM contains the latest changes. thanks again. 👍

**Comment by psaxton** [comment]
@nelsonic Thank you. 

I realized I never pushed my whitespace changes requested earlier. I have changed the test to assert on the signer type rather than truthiness as well.

I'd be happy to update the docs, though I'm not sure what to say as to me this fixes what I feel is expected behavior.

**Pull request merged**